OBPIH-6427 Location chooser test cases

kkrawczyk123 opened the pull request:
@drodzewicz PR is still a bit in progress as I need your help with:
- assertion on background colors on location chooser
- how to structure tests in the best possible way: they are super big now and there are a lot of repetitions. One of my ideas is to create separate files with tests for depots, as we create/delete organizations and locations and maybe separate file for wards. 
- once we figure out a better structure I will also improve using of testing data (org, location names etc)
- when it comes to wards I still have to figure out step 4th: https://pihemr.atlassian.net/browse/OBPIH-4644 
- session issue: test are working separately but not together for now, i know it is the same issue as with user test but I am not sure exactly at which moment I should try to include the same solution that you added there
- there are still a few selectors that could be improved

drodzewicz commented on src/pages/LocationListPage.ts:
```diff
@@ -0,0 +1,39 @@
+    return this.page.getByTestId('location-type-select');
+  }
+
+  getSelectLocationType(locationType: string) {
+    return this.page
+      .getByTestId('location-type-select')
+      .getByRole('listitem')
+      .getByText(locationType, { exact: true });
+  }
```
```suggestion
  getSelectLocationTypeOption(locationType: string) {
    return this.locationTypeSelect
      .getByRole('listitem')
      .getByText(locationType, { exact: true });
  }
```

drodzewicz commented on src/pages/OrganizationListPage.ts:
```diff
@@ -0,0 +1,33 @@
+  }
+
+  get searchByOrganizationNameField() {
+    return this.page.locator('#q');
```
```suggestion
    return this.page.getByRole('textbox', { name: 'Search' });
```

drodzewicz commented on src/tests/locationChooser.test.ts:
```diff
@@ -0,0 +1,763 @@
+      locationListPage,
+      createLocationPage,
+    }) => {
+      await test.step('Go to create organization page', async () => {
+        await page.goto('./dashboard');
+        await navbar.configurationButton.click();
+        await navbar.getNavItem('Organizations').click();
+        await organizationListPage.createOrganizationButton.click();
+      });
```
I noticed this in many places so this is more of a question what we want to test.

There are many places where we go to Dashboard page then use the navbar to go to certain page, fi we don;t care about using the navbar all that much we can shorten the test by going directly to organization create page.
We can implement following function in `CreateOrganizationPage.ts`

```ts
async goToPage() {
  await this.page.goto('./organization/create');
}
```

and then in the test just redirect to the create page
```ts
  await organizationListPage.goToPage();
```
what do you think?

drodzewicz commented on src/tests/locationChooser.test.ts:
```diff
@@ -0,0 +1,763 @@
+
+test.describe('Check if non manage inventory location is present in location chooser', () => {
+  const LOCATION_NAME = 'E2E-test-Depot';
+  const ORGANIZATION_NAME = '1000bulbs.com';
```
I think we should not use this existing organization and instead use the one created by the test

kkrawczyk123 commented on src/pages/EditOrganizationPage.ts:
```diff
@@ -0,0 +1,27 @@
+
+  async clickDeleteOrganization() {
+    await this.deleteOrganizationButton.click();
+    //this.page.on('dialog', (dialog) => dialog.accept());
```
because it was failing on this line, and it was also failing when I put it before. As it works without it I believe that it is not needed in this case

kkrawczyk123 commented:
@drodzewicz I added a few new tests and updated the existing with all your suggestions.

what I still need your help with:
- assertion on colors on location chooser
- I don't know why the last of the executed tests is failing because of dialog accepting the delete of location, it works well and the location is deleted and also it's not failing when tests are executed separately
